datafusion/proto/src/lib.rs

```diff
@@ -556,6 +556,11 @@ mod roundtrip_tests {
                     Field::new("a", DataType::Boolean, false),
                 ]),
             ),
+            ScalarValue::FixedSizeBinary(
+                b"bar".to_vec().len() as i32,
+                Some(b"bar".to_vec()),
+            ),
+            ScalarValue::FixedSizeBinary(0, None),
```

Comment:
I suggest adding a None case for non zero length as well

```suggestion
            ScalarValue::FixedSizeBinary(0, None),
            ScalarValue::FixedSizeBinary(5, None),
```